influxql/linear.go

```diff
@@ -0,0 +1,21 @@
+package influxql
+
+// linearFloat computes the the slope of the line between the points (previousTime, previousValue) and (nextTime, nextValue)
+// and returns the value of the point on the line with time windowTime
+// y = mx + b
+func linearFloat(windowTime, previousTime, nextTime int64, previousValue, nextValue float64) float64 {
+	m := (nextValue - previousValue) / float64(nextTime-previousTime) // the slope of the line
+	x := float64(windowTime - previousTime)                           // how far into the interval we are
+	b := previousValue
+	return m*x + b
+}
+
+// linearInteger computes the the slope of the line between the points (previousTime, previousValue) and (nextTime, nextValue)
+// and returns the value of the point on the line with time windowTime
+// y = mx + b
+func linearInteger(windowTime, previousTime, nextTime int64, previousValue, nextValue int64) int64 {
+	m := (nextValue - previousValue) / (nextTime - previousTime) // the slope of the line
```

Comment:
It seems to me like this won't give the most accurate answer. Maybe we should use a float as the slope and then round after we get the result instead of making this into an integer. For example, take the following values:

```
previousTime: 0
previousValue: 1
nextTime: 10
nextValue: 5
windowTime: 5
```

Now I would say that the value at 5 should be 3 here because the value halfway in between 1 and 5 is 3. But if I perform the math here, it seems I get a slope of zero and so I get a value of 1. If I use floats, I get the value 3.